### PR TITLE
Merge Feature/foods into develop

### PR DESCRIPTION
**Pull request by namirinz**
- Add status query param for `findAll`
- Fix missing addonInfo bug in 'findAll` & `findById`
- Add `published_at` & `deleted_at` in .dto

### CODE REVIEW

File: src/menus/menus.service.ts [hunk 1]
```diff
@@ -13,37 +13,95 @@ export class MenusService {
...
-    const script = [
-      { $match: { _id: { $exists: true } } },
-      {
-        $lookup: {
-          from: 'categories',
-          localField: 'category',
-          foreignField: '_id',
-          as: 'categoryInfo',
-        },
+  getAddonInfoScript = [
```
Comment (ishiasaka) [inline src/menus/menus.service.ts]: Is this private property?

File: src/menus/menus.service.ts [hunk 2]
```diff
@@ -13,37 +13,95 @@ export class MenusService {
...
-          'menus.categoryInfo': 0,
+    },
+    {
+      $project: {
+        addonInfo: 0, // Remove the addonInfo field
+      },
+    },
+  ];
+
+  groupCategoryScript = [
```
Comment (ishiasaka) [inline src/menus/menus.service.ts]: Is this private property?

File: src/menus/menus.service.ts (hunk 1 above)
Comment (namirinz) [reply src/menus/menus.service.ts]: Oh yeah

File: src/menus/menus.service.ts (hunk 2 above)
Comment (namirinz) [reply src/menus/menus.service.ts]: Done

File: src/menus/menus.service.ts (hunk 1 above)
Comment (namirinz) [reply src/menus/menus.service.ts]: Done

**Comment by namirinz** [comment]
Just added `/publish` endpoint @hirasawaau @phusitsom

File: src/menus/menus.controller.ts [hunk 3]
```diff
@@ -106,4 +110,17 @@ export class MenusController {
...
+
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description: 'The menu has been successfully published.',
+  })
+  @ApiResponse({
+    status: HttpStatus.NOT_FOUND,
+    description: 'No menu found',
+  })
+  @Patch(':id')
```
Comment (lazykern) [inline src/menus/menus.controller.ts]: Should be ':id/publish'

Comment (lazykern) [reply src/menus/menus.controller.ts]: Also don't forget to update the endpoint in tests after resolving this

**Review by lazykern — CHANGES_REQUESTED** [review]

**Comment by phusitsom** [comment]
For [category field in create menu dto](https://github.com/meeorder/meeorder-backend/blob/f44f0a4421f6a44090d8b204d75b482b44b02cea/src/menus/dto/menus.createMenu.dto.ts#L17-L19C9), the type should be string. Because it will make swagger assume that the field is an object instead of a string.
image

**Review by lazykern — COMMENTED** [review]
For the category field in create menu dto, the type should be a string. Because it will make Swagger assume that the field is an object instead of a string.

File: src/menus/menus.controller.ts (hunk 3 above)
Comment (namirinz) [reply src/menus/menus.controller.ts]: Done

**Comment by namirinz** [comment]
> For the category field in create menu dto, the type should be a string. Because it will make Swagger assume that the field is an object instead of a string.
> 
> 

Done in [ff4d836](https://github.com/meeorder/meeorder-backend/pull/31/commits/ff4d836e3ad42aafbd8cf94da9a7ef59a9b5eb75)

**Comment by lazykern** [comment]
Also, please add string property type for the category field in [this file](https://github.com/meeorder/meeorder-backend/blob/4632385cd7eb55d90b49ab1e405a8a9731148470/src/menus/dto/menus.getMenuByIdReponse.dto.ts#L19-L20) too

**Comment by lazykern** [comment]
Make sure to add swagger decorators to the `_id` field in the [schema definition](https://github.com/meeorder/meeorder-backend/blob/4632385cd7eb55d90b49ab1e405a8a9731148470/src/schema/menus.schema.ts#L9)

For example (from session schema): 
```typescript
  @Prop({ auto: true })
  @ApiProperty({ type: String, description: 'Session ID' })
  _id: Types.ObjectId;
```

Also, set the property type for those with object id type in the file as well

**Comment by namirinz** [comment]
Done in [08952a0](https://github.com/meeorder/meeorder-backend/pull/31/commits/08952a0010d781c6389b0beccd1858cec20c2057)

**Review by lazykern — DISMISSED** [review]

**Review by lazykern — APPROVED** [review]

**Comment by lazykern** [comment]
otsukaresama

**Pull request merged**